docs/spectator/lang/cpp/usage.md

```diff
@@ -1,112 +1,291 @@
-    // variable set to the Counter.
-    auto cnt_id = request_count_id_
-        ->WithTag("country", request.country)
-        ->WithTag("status", std::to_string(res.status));
-    registry_->GetCounter(std::move(cnt_id))->Increment();
-    request_latency_->Record(std::chrono::steady_clock::now() - start);
-    response_size_->Record(res.size);
-    return res;
-  }
-
- private:
-  spectator::Registry* registry_;
-  std::shared_ptr<spectator::Id> request_count_id_;
-  std::shared_ptr<spectator::Timer> request_latency_;
-  std::shared_ptr<spectator::DistributionSummary> response_size_;
-};
-
-Request get_next_request() {
-  return Request{"US"};
+{% raw %}
+
+```cpp
+#include <registry.h>
+
+int main()
+{
+    // Create common tags to be applied to all metrics sent to Atlas
+    std::unordered_map<std::string, std::string> commonTags{{"platform", "my-platform"}, {"process", "my-process"}};
+
+    // Create a config which defines the way you send metrics to SpectatorD
+    auto config = Config(WriterConfig(WriterTypes::Memory), commonTags);
+    
+    // Initialize the Registry with the Config (Always required before sending metrics)
+    auto r = Registry(config);
+
+    // Create some meters
+    auto threadGauge = r.CreateGauge("threads");
+    auto queueGauge = r.CreateGauge("queue-size", {{"my-tags", "bar"}});
+
+    threadGauge.Set(GetNumThreads());
+    queueGauge.Set(GetQueueSize());
+
+    /* Metrics Sent: 
+        "g:threads,platform=my-platform,process=my-process:5.000000\n"
+        "g:queue-size,my-tags=bar,platform=my-platform,process=my-process:10.000000\n"
+    */
 }
+```
+
+{% endraw %}
+
+## Logging
+
+Logging uses the `spdlog` library and outputs to standard output by default, with a default log
+level of `spdlog::level::info`. The `Logger` class is a singleton and provides the
+`Logger::GetLogger()` function to access the logger instance. To change the log level, call
+`Logger::GetLogger()->set_level(spdlog::level::debug);` after the logger has been successfully
+created.
+
+## Runtime Metrics
+
+Coming Soon
+
+## Working with MeterId Objects
+
+Each metric stored in Atlas is uniquely identified by the combination of the name and the tags
+associated with it. In `spectator-cpp`, this data is represented with `MeterId` objects, created
+by the `Registry`. The `CreateNewId()` method returns new a `MeterId` object, which have extra
+common tags applied, and which can be further customized by calling the `WithTag()` and
+`WithTags()` methods. Each `MeterId` will create and store a validated subset of the `spectatord`
+protocol line to be written for each `Meter`, when it is instantiated. Manipulating the tags with
+the provided methods will create new `MeterId` objects.
+
+Note that **all tag keys and values must be strings.** For example, if you want to keep track of
+the number of successful requests, then you must cast integers to strings. The `Id` class will
```

Comment:
Isn't it MeterId class? Does the Id class validate these in this language? It is 100% necessary to do this validation in Python, given that it is a dynamic language, which means that people can pass whatever they want to a given function. With C++ being strongly typed, this extra validation does not need to be done.